response_operations_ui/controllers/party_controller.py

```diff
@@ -125,6 +130,48 @@ def search_respondent_by_email(email):
     return response.json()
 
 
+def search_respondents(first_name, last_name, email_address, page=0):
+    if bool(os.environ.get('FAKE_PARTY_DATA', False)):
+        logger.debug('Respondent Search: Using fake data for controller')
+        data = mock_search_respondents(first_name, last_name, email_address, page)
+    else:
+        url = _get_search_respondents_url(first_name=first_name,
+                                          last_name=last_name,
+                                          email_address=email_address,
+                                          page=page)
+        response = requests.get(url, auth=app.config['PARTY_AUTH'])
+        if response.status_code != 200:
+            raise SearchRespondentsException(response,
```

Comment:
Is a new exception type necessary?  These details will be in the url (via parameters), so an APIError is probably more correct to throw here.

Reply:
The new exception type allows me to include data I otherwise couldn't - namely the source and email address tried to search.  This seems like it _might_ make debugging easier.

Also follows the convention of other new exceptions, like `UpdateContactDetailsException`